HW_3/hw03.Rmd

```diff
@@ -0,0 +1,170 @@
+green_data <- readr::read_csv("green_data.csv")
+library(dplyr) 
+library(ggplot2)
+library(tidyverse)
+``` 
+
+For your assignment, do the following.
+
+  1. Inspect the item responses (e.g., with graphs or by summarizing distinct values). 
+     Is anything unusual?
+```{r}
+summary(green_data) 
+
+```
+  
+ 
+  2. Compute total scores for the four scales. 
+     Recode variables as needed.
+```{r}
+green_data %>%
+mutate(
+  across(eeverything(),
+ ~ recode(.x , "-99" = NA, .default = .x)
+))
+
+#   ~ says that you are going to use the .x symbol , .x marks the spot 
+# 'everything()' = ALL COLUMNS 
+# .missing is for when you look for missing values 
+# NA is for when this is missing 
+# .x = the current column 
+
+```   
+  
+  
+  3. Rescale the variables so that they go from 0-100 instead of the original range.
+     Name the rescaled variables `*_pomp`.
+```{r}
+reversed <- c("green1", "green2","green3", "green4", 
+              "green5" , "comp1", "comp2", "comp3", 
+              "comp4", "comp5", "comp6", "comp7", "comp8", "comp9", "comp10", 
+              "intel1", "intel2", "intel3", "intel4", "intel5", "intel6",
+              "intel7", "intel8", "intel9", "intel10", "open1",
+              "open2", "open3", "open4", "open5", "open6", "open7", "open8",
+              "open9", "open10")
+new_green_data <- green_data %>% 
+                    mutate(
+                           across(all_of(reversed),
+                           ~ recode(.x, "1" = 0, "2" = 25, "4" = 75, "5" = 100)
+                          )) %>% 
+  rowwise() %>% 
+  mutate(green_pomp = sum(c
+                         (green1,green2,green3,green4,green5), na.rm = TRUE),
+         comp_pomp = sum(c
+                        (comp1,comp2,comp3,comp4,comp5,comp6,comp7,comp8,comp9,comp10), na.rm = TRUE),
+         intel_pomp = sum(c
+                         (intel1,intel2,intel3,intel4,intel5,intel6,intel7,intel8,intel9,intel10), na.rm = TRUE),
+         open_pomp = sum(c
+                        (open1,open2,open3,open4,open5,open6,open7,open8,open9,open10), na.rm = TRUE),) 
+
+```
+ 
+ 
+ 
+ 
+  4. Make plots that illustrate the distributions of the 4 POMP-scored variables.
+  
+```{r}
+ggplot(new_green_data) +
+  aes(x = green_pomp) +
+  geom_histogram( fill = "pink")+
+  theme_classic()
+
+ggplot(new_green_data) +
+  aes(x =intel_pomp) +
+  geom_histogram(fill = "green")+
+  theme_classic()
+
+ggplot(new_green_data) +
+  aes(x = comp_pomp) +
+  geom_histogram(fill = "blue")+
+  theme_classic()
+
+ggplot(new_green_data) +
+  aes(x = open_pomp) +
+  geom_histogram(fill = "purple")+
+  theme_classic()
+
+```  
+  
+  
+  
+  5. Make scatterplots showing the relationships between **green reputation** and each personality trait.
+     Include trend lines for **students** and **non-students**.
+     What do these plots show?
+```{r}
+
+new_green_data %>%
+  mutate(
+    student = recode(student , "1" = "Student" , "2" = "NonStudent")
+  )
+
```

Comment:
It looks like there's a -99 in the student column from the last plot. I think this *might* be because you didn't specify what the recode command should do if the values are anything other than a 1 or a 2?

Here's the code I used:

```
student = recode(student, "1" = "not a student", "2" = "student", .missing = "(Unknown)"
```

Reply:
That makes sense. Thank you.